pkg/cmd/server/origin/storage.go

```diff
@@ -0,0 +1,548 @@
+	policyetcd "github.com/openshift/origin/pkg/authorization/registry/policy/etcd"
+	policybindingregistry "github.com/openshift/origin/pkg/authorization/registry/policybinding"
+	policybindingetcd "github.com/openshift/origin/pkg/authorization/registry/policybinding/etcd"
+	"github.com/openshift/origin/pkg/authorization/registry/resourceaccessreview"
+	rolestorage "github.com/openshift/origin/pkg/authorization/registry/role/policybased"
+	rolebindingstorage "github.com/openshift/origin/pkg/authorization/registry/rolebinding/policybased"
+	rolebindingrestrictionetcd "github.com/openshift/origin/pkg/authorization/registry/rolebindingrestriction/etcd"
+	"github.com/openshift/origin/pkg/authorization/registry/selfsubjectrulesreview"
+	"github.com/openshift/origin/pkg/authorization/registry/subjectaccessreview"
+	"github.com/openshift/origin/pkg/authorization/registry/subjectrulesreview"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	securityapiv1 "github.com/openshift/origin/pkg/security/api/v1"
+	"github.com/openshift/origin/pkg/security/registry/podsecuritypolicyreview"
+	"github.com/openshift/origin/pkg/security/registry/podsecuritypolicyselfsubjectreview"
+	"github.com/openshift/origin/pkg/security/registry/podsecuritypolicysubjectreview"
+	oscc "github.com/openshift/origin/pkg/security/scc"
+
+	// register api groups
+	_ "github.com/openshift/origin/pkg/api/install"
+)
+
+// TODO this function needs to be broken apart with each API group owning their own storage, probably with two method
+// per API group to give us legacy and current storage
+func (c OpenshiftAPIConfig) GetRestStorage() (map[schema.GroupVersion]map[string]rest.Storage, error) {
+	// TODO sort out who is using this and why.  it was hardcoded before the migration and I suspect that it is being used
+	// to serialize out objects into annotations.
+	externalVersionCodec := kapi.Codecs.LegacyCodec(schema.GroupVersion{Group: "", Version: "v1"})
+
+	//TODO/REBASE use something other than c.KubeClientsetInternal
+	nodeConnectionInfoGetter, err := kubeletclient.NewNodeConnectionInfoGetter(c.KubeClientExternal.CoreV1().Nodes(), *c.KubeletClientConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to configure the node connection info getter: %v", err)
+	}
+
+	// TODO: allow the system CAs and the local CAs to be joined together.
+	importTransport, err := restclient.TransportFor(&restclient.Config{})
+	if err != nil {
+		return nil, fmt.Errorf("Unable to configure a default transport for importing: %v", err)
+	}
+	insecureImportTransport, err := restclient.TransportFor(&restclient.Config{
+		TLSClientConfig: restclient.TLSClientConfig{
+			Insecure: true,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Unable to configure a default transport for importing: %v", err)
+	}
+
+	buildStorage, buildDetailsStorage, err := buildetcd.NewREST(c.GenericConfig.RESTOptionsGetter)
+	if err != nil {
```

Comment:
> would write those if-clause in one line, if gofmt accepts that. Func gets very long with 50% error handling.

I miss java too :)

This function is on the chopping block.  I wouldn't get overly caught up in it.